test/test_evaluators.py

```diff
@@ -839,3 +839,45 @@ def test_l2_evaluator(self):
         evaluator_l2.add_predictions(predictions=self.PREDICTIONS[5:], targets=self.TARGETS[5:])
         report = evaluator_l2.get_report()
         self.assertAlmostEqual(report[evaluator_l2._get_id()], np.sqrt(10) / 10, places=4)
+
+
+class TestConfusionMatrixEvaluator(unittest.TestCase):
+    TARGETS = [np.array(["l1", "l2", "l3", "l4", "l5", "l3", "l3", "l2", "l1", "l0"]),
+               np.array(["l1", "l1", "l1", "l4", "l3", "l3", "l3", "l2", "l1", "l1"]),
+               np.array([0])]
+    PREDICTIONS = [np.array(["l0", "l1", "l2", "l3", "l4", "l5", "l3", "l3", "l3", "l1"]),
+                   np.array(["l0", "l1", "l2", "l3", "l4", "l2", "l3", "l3", "l3", "l1"]),
+                   np.array([1])]
+    CM_GT = [np.array([[0, 1., 0., 0., 0., 0.],
+                      [0.5, 0., 0., 0.5, 0., 0.],
+                      [0., 0.5, 0., 0.5, 0., 0.],
+                      [0., 0.0, 0.33333, 0.33333, 0., 0.33333],
+                      [0., 0., 0., 1., 0., 0.],
+                      [0., 0., 0., 0., 1., 0.]]),
+             np.array([[0, 0, 0, 0, 0],
+                       [1, 2, 1, 1, 0],
+                       [0, 0, 0, 1, 0],
+                       [0, 0, 1, 1, 1],
+                       [0, 0, 0, 1, 0]]),
+             np.array([[0, 1],
+                       [0, 0]])]
+
+    LABELS = [["l0", "l1", "l2", "l3", "l4", "l5"],
+              ["l0", "l1", "l2", "l3", "l4"],
+              [0, 1]]
+
+    NORMALIZATIONS = [True, False, False]
+
+    def test_confusion_matrix_evaluator(self):
+        evaluator_conf_mat = ConfusionMatrixEvaluator()
+        for idx in range(len(self.TARGETS)):
```

Comment:
some more tests would help:
1. get_report is called before any predictions/gts are added (aka 0 class)
2. only 1 class
3. labels out of range
4. eval1, reset(), rerun eval1, assert result(eval1) == result(rerun_eval1)